pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+                # can stop looking through whole list.
+
+        if not rtn_inst:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Instance %s not found in '
+                                              'repository' % iname)
+
+        if 'PropertyList' in params:
+            property_list = params['PropertyList']
+            if property_list == "":
+                rtn_inst.properties = NocaseDict()
+            elif property_list is None:
+                pass
+            elif isinstance(property_list, list):
+                new_properties = NocaseDict()
+                properties = rtn_inst.properties
+                for pn in property_list:
+                    if not isinstance(pn, six.string_types):
+                        raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                                       'Property name %s in  '
+                                       'PropertyList %s invalid'
+                                       % (pn, property_list))
+
+                    if pn in properties:
+                        new_properties[pn] = properties[pn]
+                rtn_inst.properties = new_properties
+            else:
+                raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                               'propertyList %s invalid' %
+                               property_list)
+
+        if 'IncludeQualifiers' in params and not ['IncludeQualifiers']:
+            self._remove_qualifiers(rtn_inst)
+
+        if 'IncludeClassOrigin' in params and not params['IncludeClassOrigin']:
+            self._remove_classorigin(rtn_inst)
+
+        return rtn_inst
+
+    def _enumerate_instances(self, classname, namespace, **params):
+        """
+        Local function to enumerate all instances that match classname in
+        namespace. Gets all instances in classname and subclasses of classname
+        """
+        insts = []
+
+        classes = self._get_subclass_names(classname, namespace) \
+            if self.classes else []
+
+        classes.append(classname)
+        try:
+            insts = self.instances[namespace]
+            inst_list = [self._get_instance(inst.path, namespace, **params)
+                         for inst in insts if inst.path.classname in classes]
+        except KeyError:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE, '%s' % namespace)
+
+        return inst_list
+
+    def _enumerate_instancenames(self, classname, namespace):
+        """
+        Local function to enumerate all instance names that match classname in
+        namespace. Gets all instanceNames in classname and subclasses of
+        classname
+        """
+        insts = []
+        classes = self._get_subclass_names(classname, namespace) \
+            if self.classes else []
+
+        classes.append(classname)
+        try:
+            insts = self.instances[namespace]
+            inst_paths = [copy.copy(inst.path) for inst in insts
+                          if inst.path.classname in classes]
```

Comment:
The try-block should include just the `insts = ...` statement.

Reply:
Done